components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -229,6 +237,24 @@ public void buildIntermediateCertValidationConfigData() {
         }
     }
 
+    /**
+     * Build allowedTenantDomains by splitting the allowedTenantDomains string.
+     *
+     * @param crossAccessAllowedTenants allowedTenantDomains.
+     * @return List of allowedTenantDomains.
+     */
+    public List<String> crossAccessAllowedTenants(String crossAccessAllowedTenants) {
+
+        if (StringUtils.isNotBlank(crossAccessAllowedTenants)) {
+            List<String> allowedTenantDomainsList = new ArrayList<>();
+            String regex = "\\s*,\\s*";
+            String[] allowedTenantDomainsNames = crossAccessAllowedTenants.split(regex);
+            allowedTenantDomainsList.addAll(Arrays.asList(allowedTenantDomainsNames));
+            return allowedTenantDomainsList;
+        }
+        return null;
+    }
+
```

Comment:
This has only a single usage, right? If that's the case do we need to provide this as a public method? Also, we can pass any string with `,` and get a list. So we cannot say the returned list is a tenant list. WDYT?

Reply:
I don't think this is something to be given as public util method